File: Archiver/app_code/clsFileInfo.cs
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;
using MODI;

namespace EcmArchiver
{
    public class clsFileInfo
    {
        private string FileKey = "";

        public void RemoveBadChars(ref string FQN)
        {
            for (int i = 1, loopTo = FQN.Length; i <= loopTo; i++)
            {
                string CH = Strings.Mid(FQN, i, 1);
                switch (CH ?? "")
                {
                    case "/":
                        {
                            StringType.MidStmtStr(ref FQN, i, 1, ".");
                            break;
                        }

                    case "'":
                        {
                            StringType.MidStmtStr(ref FQN, i, 1, ".");
                            break;
                        }

                    case " ":
                        {
                            StringType.MidStmtStr(ref FQN, i, 1, ".");
                            break;
                        }

                    case ":":
                        {
                            StringType.MidStmtStr(ref FQN, i, 1, ".");
                            break;
                        }
                }
            }
        }
    }
}

File: Archiver/app_code/clsFileInfo.vb
Imports System.IO

Public Class clsFileInfo

Dim FileKey AS String  = "" 

    Sub RemoveBadChars(ByRef FQN AS String )
        For i As Integer = 1 To FQN.Length
Dim CH AS String  = Mid(FQN, i, 1) 
            Select Case CH
                Case "/"
                    Mid(FQN, i, 1) = "."
                Case "'"
                    Mid(FQN, i, 1) = "."
                Case " "
                    Mid(FQN, i, 1) = "."
                Case ":"
                    Mid(FQN, i, 1) = "."
            End Select
        Next
    End Sub
End Class
